### PR TITLE
change strategy to recreate if volumes present, fix #264

### LINKED ISSUE #264: Change strategy for Deployments/DeployementConfigs
When Kompose creates Deployments there is no strategy specified -  RollingUpdate is default.

This might cause problems with some applications using volumes, like databases.

We should consider changing strategy to Recreate, this should be safer option.

### PR DESCRIPTION
**Pull request by containscafeine**

### CODE REVIEW

**Comment by surajssd** [comment]
@containscafeine are you still working on this?

**Comment by containscafeine** [comment]
@surajssd yep

**Comment by containscafeine** [comment]
@kadel @surajssd @cdrage up for review, PTAL!

File: pkg/transformer/kubernetes/k8sutils.go [hunk 1]
```diff
@@ -384,6 +384,15 @@ func (k *Kubernetes) UpdateKubernetesObjects(name string, service kobject.Servic
...
 		k.UpdateController(obj, fillTemplate, fillObjectMeta)
+
+		if len(service.Volumes) > 0 {
+			switch objType := obj.(type) {
+			case *extensions.Deployment:
+				objType.Spec.Strategy.Type = extensions.RecreateDeploymentStrategyType
+			case *deployapi.DeploymentConfig:
+				objType.Spec.Strategy.Type = deployapi.DeploymentStrategyTypeRecreate
+			}
+		}
```
Comment (cdrage) [inline pkg/transformer/kubernetes/k8sutils.go]: LGTM :+1:

**Comment by surajssd** [comment]
code LGTM, but wanna give it a try once by running it against application!

**Comment by surajssd** [comment]
@containscafeine also title says 'no test' 

*change strategy to recreate if volumes present, no tests, fix #264*

but you seem to have added tests?

**Comment by surajssd** [comment]
@containscafeine tested with etherpad application from fixtures and it works! would like to have this behavior documented then we are good to merge this.

**Comment by containscafeine** [comment]
@surajssd thanks for pointing that out, added docs.

**Comment by containscafeine** [comment]
@surajssd good to go now?

**Pull request merged**